test/unit/math/mix/core/operator_minus_minus_test.cpp

```diff
@@ -1,26 +1,36 @@
-#include <test/unit/math/mix/mat/util/autodiff_tester.hpp>
+#include <test/unit/math/test_ad.hpp>
 
-struct op_minus_minus_pre_f {
-  template <typename T1, typename T2>
-  static typename boost::math::tools::promote_args<T1, T2>::type apply(
-      const T1& x1, const T2& x2) {
-    typename boost::math::tools::promote_args<T1, T2>::type y = x1;
-    return --y;
-  }
-};
-
-struct op_minus_minus_post_f {
-  template <typename T1, typename T2>
-  static typename boost::math::tools::promote_args<T1, T2>::type apply(
-      const T1& x1, const T2& x2) {
-    typename boost::math::tools::promote_args<T1, T2>::type y = x1;
-    return y--;
-  }
-};
+TEST(mathMixCore, opratorMinusMinusPre1) {
+  auto f = [](const auto& x1) {
+    auto y = x1;
+    auto z = --y;
+    return y;
+  };
+  stan::test::expect_common_unary(f);
+}
+TEST(mathMixCore, opratorMinusMinusPre2) {
```

Comment:
Because `x1` is const.  So we need a local variable we can apply `--` to and then return the value.  This first functor returns the value before and the second one the value after.  I'll add some doc.